feature:

LIS-One opened the pull request:
-added controller for customer worker ticker
-added configuration for creating webclient for worker and customer -added service and service impl to make a requests to this services

LIS-One committed:
feature:
-added controller for customer worker ticker
-added configuration for creating webclient for worker and customer
-added service and service impl to make a requests to this services

AglaiaHub commented on dashboard-service-api/src/main/java/rent/vehicle/dashboardserviceapi/controller/DashboardController.java:
Я бы лучше назвала его воркер контроллер. Потому что весь сервис дашбоард, так будет проще понимать что мы тут контроллим

AglaiaHub commented on dashboard-service-model/src/main/java/rent/vehicle/enums/CustomerRole.java:
Почему роли такие? Вроде мы обсуждали что роли будут Технай Саппортер и Админ который бог

AglaiaHub commented on dashboard-service-api/pom.xml:
<dependency>
			<groupId>rent.vehicle</groupId>
			<artifactId>device-service-model</artifactId>
			<version>1.3-SNAPSHOT</version>
			<artifactId>worker-service-model</artifactId>
			<version>0.0.1-SNAPSHOT</version>
		</dependency> вот эта депенденси у меня помнялась. Не знаю, как нам теперь ее изменить и в твоем коде??

AglaiaHub replied:
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
    <modelVersion>4.0.0</modelVersion>
    <parent>
        <groupId>org.springframework.boot</groupId>
        <artifactId>spring-boot-starter-parent</artifactId>
        <version>3.5.3</version>
        <relativePath/> <!-- lookup parent from repository -->
    </parent>
    <groupId>rent.vehicle</groupId>
    <artifactId>dashboard-service-api</artifactId>
    <version>0.0.1-SNAPSHOT</version>
    <name>dashboard-service-api</name>
    <description>dashboard-service-api</description>
    <url/>
    <licenses>
        <license/>
    </licenses>
    <developers>
        <developer/>
    </developers>
    <scm>
        <connection/>
        <developerConnection/>
        <tag/>
        <url/>
    </scm>
    <properties>
        <java.version>21</java.version>
    </properties>
    <dependencies>
        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-web</artifactId>
        </dependency>

        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-devtools</artifactId>
            <scope>runtime</scope>
            <optional>true</optional>
        </dependency>
        <dependency>
            <groupId>org.projectlombok</groupId>
            <artifactId>lombok</artifactId>
            <optional>true</optional>
        </dependency>

        <dependency>
            <groupId>com.fasterxml.jackson.datatype</groupId>
            <artifactId>jackson-datatype-jsr310</artifactId>
        </dependency>

        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-webflux</artifactId>
        </dependency>

        <!-- https://mvnrepository.com/artifact/org.springframework.data/spring-data-jpa -->
        <dependency>
            <groupId>org.springframework.data</groupId>
            <artifactId>spring-data-jpa</artifactId>
        </dependency>

        <dependency>
            <groupId>org.modelmapper</groupId>
            <artifactId>modelmapper</artifactId>
            <version>3.2.2</version>
        </dependency>

        <!-- dto, enum, exceptions -->
        <dependency>
            <groupId>rent.vehicle</groupId>
            <artifactId>dashboard-service-model</artifactId>
            <version>1.0-SNAPSHOT</version>
        </dependency>

        <dependency>
            <groupId>junit</groupId>
            <artifactId>junit</artifactId>
            <scope>test</scope>
        </dependency>

        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-test</artifactId>
            <scope>test</scope>
        </dependency>
    </dependencies>

    <build>
        <plugins>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-compiler-plugin</artifactId>
                <configuration>
                    <annotationProcessorPaths>
                        <path>
                            <groupId>org.projectlombok</groupId>
                            <artifactId>lombok</artifactId>
                        </path>
                    </annotationProcessorPaths>
                </configuration>
            </plugin>
            <plugin>
                <groupId>org.springframework.boot</groupId>
                <artifactId>spring-boot-maven-plugin</artifactId>
                <configuration>
                    <excludes>
                        <exclude>
                            <groupId>org.projectlombok</groupId>
                            <artifactId>lombok</artifactId>
                        </exclude>
                    </excludes>
                </configuration>
            </plugin>
        </plugins>
    </build>

</project>

AglaiaHub commented on dashboard-service-api/src/main/java/rent/vehicle/dashboardserviceapi/configuration/DashboardConfiguration.java:
Это надо тоже сверить с моим и сделать так, чтобы ни мой ни твой код не поломались. Не знаю как

AglaiaHub commented on dashboard-service-api/src/main/java/rent/vehicle/dashboardserviceapi/configuration/QueryParamUtil.java:
аналогично - сверить с моим кодом и проверить, чтобы ни мой ни твой код не поломались

AglaiaHub commented on dashboard-service-model/src/main/java/rent/vehicle/common/CustomPage.java:
аналогично, сверить с моим кодом и проверить, чтобы ни мой ни твой код не поломались

AglaiaHub reviewed (changes requested):
Привет! Я проверила и у меня там пара вопросов по классам, я только не понимаю, что теперь с этим делать и как сверить, что у нас код не противоречит

LIS-One committed:
feature:
-refactored a lot of stuff e.g. created separate packages for each service or entity as needed
-added SearchAdapterService to convert text from uri to GenericSearchRequest
-added all must have endpoints

LIS-One committed:
feature:
-added all endpoints
-configured controller methods with @modelAttribute
-checked and changed poms according to master branch

LIS-One committed:
Merge branch 'master' into ivan-customer-worker-integration
